### ISSUE TITLE
UnicodeDecodeError or BadGzipFile error is raised when initializing a client

### ISSUE THREAD

**Issue by zhong-yy** [obspy/obspy#3130]
### Avoid duplicates

- [X] I searched existing issues

### Bug Summary

I use multiprocessing package to download data in parallel. Sometimes, it works well, but sometimes I get a `UnicodeDecodeError` or `gzip.BadGzipFile` error from client initialization.

### Code to Reproduce

```python
import multiprocessing as mp
from obspy.clients.fdsn import Client
from obspy.clients.fdsn.header import FDSNNoServiceException
import time


def test_fun():
    while True:
        try:
            client = Client("IRIS", timeout=600)
        except FDSNNoServiceException:
            print("FDSNNoServiceException. Waiting for 30s before next try ...")
            time.sleep(30)
            continue
        else:
            print(f"{mp.current_process().name}: Client is initialized successfully")
            break


def run_parallel(num_processes):
    processes = []
    for _ in range(num_processes):
        proc = mp.Process(target=test_fun)
        processes.append(proc)

    for proc in processes:
        proc.start()  # start processes
        print(f"{proc.name} started")

    # wait for all processes to complete
    for proc in processes:
        print(f"Is {proc.name} alive? {proc.is_alive()}")
        proc.join()
        print(f"Finished joining {proc.name}")


if __name__ == "__main__":
    run_parallel(16)
```

### Error Traceback

```Python traceback
1. UnicodeDecodeError

Process Process-10:
Traceback (most recent call last):
  File "/home/zyy/miniconda3/envs/obspy/lib/python3.10/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/home/zyy/miniconda3/envs/obspy/lib/python3.10/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/home/zyy/test.py", line 10, in test_fun
    client = Client("IRIS", timeout=600)
  File "/home/zyy/miniconda3/envs/obspy/lib/python3.10/site-packages/obspy/clients/fdsn/client.py", line 276, in __init__
    self._discover_services()
  File "/home/zyy/miniconda3/envs/obspy/lib/python3.10/site-packages/obspy/clients/fdsn/client.py", line 1517, in _discover_services
    decoded_wadl = wadl.decode('utf-8')
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x8b in position 1: invalid start byte
```
2. gzip.BadGzipFile
```
Exception in thread Thread-3:
Traceback (most recent call last):
  File "/home/zyy/miniconda3/envs/obspy/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/home/zyy/miniconda3/envs/obspy/lib/python3.10/site-packages/obspy/clients/fdsn/client.py", line 1476, in run
    code, data = download_url(
  File "/home/zyy/miniconda3/envs/obspy/lib/python3.10/site-packages/obspy/clients/fdsn/client.py", line 1856, in download_url
    data = f.read()
  File "/home/zyy/miniconda3/envs/obspy/lib/python3.10/gzip.py", line 301, in read
    return self._buffer.read(size)
  File "/home/zyy/miniconda3/envs/obspy/lib/python3.10/_compression.py", line 118, in readall
    while data := self.read(sys.maxsize):
  File "/home/zyy/miniconda3/envs/obspy/lib/python3.10/gzip.py", line 488, in read
    if not self._read_gzip_header():
  File "/home/zyy/miniconda3/envs/obspy/lib/python3.10/gzip.py", line 436, in _read_gzip_header
    raise BadGzipFile('Not a gzipped file (%r)' % magic)
gzip.BadGzipFile: Not a gzipped file (b'a\r')
```
```


### ObsPy Version?

1.3.0

### Operating System?

Ubuntu 22

### Python Version?

3.10.5

### Installation Method?

conda

**Comment by megies** [comment]
Do you really need multiprocessing with IRIS FDSNWS, I would assume that download speed is the limiting factor if you do a bulk request

**Comment by zhong-yy** [comment]
> IRIS FDSNWS

Hi, I think multiprocessing can reduce the downloading time. I am not sure whether bulk request can download data in parallel. 

In addition, I found that `get_waveforms_bulk` cannot catch FDSNNoDataError and some network related errors (e.g. IncompleteRead, ValueError, FDSNException), so I decided to use `get_waveforms` to loop over a list of requests and keep track of failed requests.

**Comment by megies** [comment]
You should really use bulk request, individual requests create a massive http overhead. If there is anything that makes bulk request inconvenient to use, we should rather address those things

**Comment by zhong-yy** [comment]
> You should really use bulk request, individual requests create a massive http overhead. If there is anything that makes bulk request inconvenient to use, we should rather address those things

I have opened another issue about the bulk request. 
https://github.com/obspy/obspy/issues/3131#issue-1343983312

**Comment by zhong-yy** [comment]
Is there any way to catch the exception in thread?

**Comment by megies** [comment]
see #3131 i think you should just do a bulk request and then compare what data you did not get
